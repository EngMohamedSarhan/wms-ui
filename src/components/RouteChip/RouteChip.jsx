import { Chip } from "@mui/material";
import PropTypes from "prop-types";
import React, { createElement } from "react";

const RouteChip = ({ label, icon, sx, ...props }) => (
  <Chip
    color="primary"
    variant="outlined"
    icon={createElement(icon, { fontSize: "small" })}
    label={label}
    sx={{ fontWeight: "500", ...sx }}
    {...props}
  />
);

RouteChip.propTypes = {
  label: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  sx: PropTypes.object,
};

export default RouteChip;
