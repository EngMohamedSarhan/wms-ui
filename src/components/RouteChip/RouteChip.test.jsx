/* eslint-disable no-undef */
import { TextSnippet } from "@mui/icons-material";
import { render } from "@testing-library/react";
import React from "react";

import RouteChip from "./RouteChip";

describe("RouteChip Component", () =>
  test("RouteChip render", () => {
    const { getByText } = render(<RouteChip label="test" icon={TextSnippet} />);
    const chip = getByText("test");
    expect(chip).toBeInTheDocument();
  }));
