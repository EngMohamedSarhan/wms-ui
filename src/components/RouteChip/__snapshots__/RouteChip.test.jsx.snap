// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteChip Component RouteChip snapshot 1`] = `
<ForwardRef(Chip)
  color="primary"
  data-testid="route-chip"
  icon={
    <Memo
      fontSize="small"
    />
  }
  label="test"
  sx={
    Object {
      "fontWeight": "500",
    }
  }
  variant="outlined"
/>
`;
